src/Form/index.tsx

```diff
@@ -576,27 +576,34 @@ function Form({
   const getNewStep = async (newKey: any) => {
     // @ts-expect-error TS(2531): Object is possibly 'null'.
     let newStep = steps[newKey];
-    while (true) {
-      const loadCond = newStep.next_conditions.find((cond: any) => {
-        if (cond.element_type !== 'step') return false;
-        const notAuth =
-          cond.rules.find((r: any) => r.comparison === 'not_authenticated') &&
-          !initState.authId &&
-          // Re: firebaseConfirmationResult, the user hasn't authenticated yet
-          // but they're in the process of doing so and we don't want to
-          // consider them "unauthenticated" for the purposes of redirecting
-          !window.firebaseConfirmationResult;
-        const auth =
-          cond.rules.find((r: any) => r.comparison === 'authenticated') &&
-          initState.authId;
-        return notAuth || auth;
-      });
-      if (loadCond) {
-        if (changeStep(loadCond.next_step_key, newKey, steps, history)) {
-          clearLoaders(true);
-          return;
-        }
-      } else break;
+
+    const metadata =
+      integrations.stytch?.metadata ?? integrations.firebase?.metadata;
+    const authSteps = metadata?.auth_gate_steps ?? [];
+    if (authSteps.length > 0) {
+      const userAuthed = Boolean(initInfo().authId);
+      const nextStepIsProtected = authSteps.includes(newStep.id);
+      const findStepName = (stepId: string) => {
+        const step: undefined | { key: string } = Object.values(steps).find(
+          (step: any) => step.id === stepId
+        ) as undefined | { key: string };
+        return step?.key;
+      };
+      let nextStep: undefined | string;
+
+      if (userAuthed && initInfo().loginRedirectFlag) {
```

Comment:
we seem to use initInfo().x to read and initState.x = y to write
is this the desired distinction? if so, should initInfo() return a spread initState so values can't be set?

Reply:
that is the desired distinction, but spreading isn't a perfect solve since it's a shallow copy